pkg/clustermanager/cluster_manager.go

```diff
@@ -778,7 +776,17 @@ func getProviderNamespaces(providerDeployments map[string][]string) []string {
 }
 
 func (c *ClusterManager) InstallMachineHealthChecks(ctx context.Context, clusterSpec *cluster.Spec, workloadCluster *types.Cluster) error {
-	mhc, err := templater.ObjectsToYaml(kubernetes.ObjectsToRuntimeObjects(clusterapi.MachineHealthCheckObjects(clusterSpec.Cluster, c.unhealthyMachineTimeout, c.nodeStartupTimeout))...)
+	unhealthyMachineTimeout, err := time.ParseDuration(clusterSpec.Cluster.Spec.MachineHealthCheck.UnhealthyMachineTimeout)
+	if err != nil {
+		return err
+	}
+
+	nodeStartupTimeout, err := time.ParseDuration(clusterSpec.Cluster.Spec.MachineHealthCheck.NodeStartupTimeout)
+	if err != nil {
+		return err
+	}
+
+	mhc, err := templater.ObjectsToYaml(kubernetes.ObjectsToRuntimeObjects(clusterapi.MachineHealthCheckObjects(clusterSpec.Cluster, unhealthyMachineTimeout, nodeStartupTimeout))...)
```

Comment:
why wouldn't the `MachineHealthCheckObjects` simply take the timeouts from the cluster?

Reply:
it also uses the name of the cluster for generating the machine health check name.

Reply:
I think he means have the `time.ParseDuration` calls inside the `clusterapi.MachineHealthCheckObjects` right?